### ISSUE TITLE
Publisher documentation's examples are hard to parse

### ISSUE THREAD

**Issue by AdamWill** [inway/mojo-rabbitmq-client#33]
While trying to grok how this module works, I was very confused by the Publisher docs. Specifically, this bit:

"So this:

    $publisher->publish($body, { header => 'content' });

can be also written like this:

    $publisher->publish($body, header => { header => 'content' });

But beware - headers get merged, but params override values so when you write this:

    $publisher->publish({ json => 'object' }, header => { header => 'content' });

message will lack C<content_type> header!"

The third example there is very confusing. The fact that the code in that example differs from the code in the previous example strongly implies that *this difference is why the `content_type` header would be missing*, and that the `content_type` header would not be missing in the second example. But IIUC, that is not true. In fact the `content_type` header would be present for the first example, but missing for *both* the second *and* third examples. It's missing because we're passing in `header` as one of the `params`, so that overrides the `header` constructed by the Publisher code (which includes the `content_type`), not because of the syntax used to construct the body - right?

If I'm right, it'd be great if this could be made clearer...

**Comment by SPodjasek** [comment]
Released in v0.2.3
